### ISSUE TITLE
Don't render entire App whenever a buffer changes.

### ISSUE THREAD

**Issue by mbebenita** [wasdk/WebAssemblyStudio#142]
This is caused by this `forceUpdate` - https://github.com/wasdk/WebAssemblyStudio/blob/master/src/components/App.tsx#L215

We should update whenever a file's dirty status changes, not whenever the buffer changes, which happens very frequently. It's quite impressive the app is as fast as it is, given that it's refreshing on every key stroke.

**Comment by Scoombe** [comment]
@mbebenita is the fix to remove this function?

``` javascript
appStore.onDidChangeBuffer.register(() => {
      this.forceUpdate();
});
```

Or is there something to add?

**Comment by mbebenita** [comment]
I think we need to add another kind of event, like `onDidChangeMetaData` that only fires for things the UI cares about. The reason we have the `forceUpdate` for `onDidChangeBuffer` is because we want to show the `M` next to files that have been modified.

**Comment by mbebenita** [comment]
> @mbebenita is it only reason why does appStore.onDidChangeBuffer.register exist?

Yes, I think so. All other views register themselves to the file's buffer directly.

**Comment by yurydelendik** [comment]
Fixed by #195
